packages/test/local-server-stress-tests/src/localServerStressHarness.ts

```diff
@@ -1338,3 +1358,100 @@ export namespace createLocalServerStressSuite {
 				skip: [...seeds, ...(providedOptions?.skip ?? [])],
 			});
 }
+
+/**
+ * Operation type for cloning a client using its pending local state.
+ *
+ * This operation triggers a clone of an existing client by capturing its
+ * pending local state (via `getPendingLocalState()`) and loading a new
+ * container from it.
+ */
+interface GetClientPendingState {
+	type: "getClientPendingState";
+	sourceClientTag: `client-${number}`;
+	newClientTag: `client-${number}`;
+}
+
+/**
+ * Adds support for "getPendingLocalState"-based client cloning to the stress model.
+ *
+ * This mixin extends the base fuzz model by occasionally inserting operations that:
+ * - Capture a given client's pending local state.
+ * - Instantiate a new client from that state using the same container URL.
+ * - Add the cloned client to the test state for further fuzzing.
+ *
+ * Preconditions:
+ * - Must have at least one client already in the system.
+ * - Must be in an attached container state.
+ */
+function mixinGetClientPending<TOperation extends BaseOperation>(
+	model: LocalServerStressModel<TOperation>,
+	options: LocalServerStressOptions,
+): LocalServerStressModel<TOperation | GetClientPendingState> {
+	const generatorFactory: () => AsyncGenerator<
+		TOperation | GetClientPendingState,
+		LocalServerStressState
+	> = () => {
+		const baseGenerator = model.generatorFactory();
+		return async (
+			state: LocalServerStressState,
+		): Promise<TOperation | GetClientPendingState | typeof done> => {
+			const { clients, random, validationClient } = state;
+			if (
+				options.clientJoinOptions !== undefined &&
+				validationClient.container.attachState !== AttachState.Detached &&
+				clients.length > 0
+			) {
+				return {
+					type: "getClientPendingState",
+					sourceClientTag: random.pick(clients).tag,
+					newClientTag: state.tag("client"),
+				} satisfies GetClientPendingState;
+			}
+			return baseGenerator(state);
+		};
+	};
+
+	const minimizationTransforms: MinimizationTransform<TOperation | GetClientPendingState>[] =
+		(model.minimizationTransforms as
+			| MinimizationTransform<TOperation | GetClientPendingState>[]
+			| undefined) ?? [];
+
+	const reducer: AsyncReducer<
+		TOperation | GetClientPendingState,
+		LocalServerStressState
+	> = async (state, op) => {
+		if (isOperationType<GetClientPendingState>("getClientPendingState", op)) {
+			const sourceClient = state.clients.find((c) => c.tag === op.sourceClientTag);
+			assert(sourceClient !== undefined, `Client ${op.sourceClientTag} not found`);
+
+			assert(
+				typeof sourceClient.getPendingLocalState === "function",
+				`Client ${op.sourceClientTag} does not have pending local state`,
+			);
+			const pendingLocalState = await sourceClient.getPendingLocalState?.();
+
+			const url = await state.validationClient.container.getAbsoluteUrl("");
+			assert(url !== undefined, "url of container must be available");
+			const newClient = await loadClient(
```

Comment:
this reducer should also close the existing client. It's invalid usage to do this sort of thing with pending local state.

Reply:
I think longer-term @anthony-murphy is wanting to permit loading a new client in a readonly state without closing the existing client.  But agree that closing it is at minimum the right thing to do for now.